docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,1052 @@
+      tags:
+        version: v1
+    conf:
+      action: DENY
+```
+
+Now we want to figure
+what configuration we'll get on "backend" data plane proxy.
+
+Merge all "from" sections from all policies
+matched for "backend" proxy:
+
+```yaml
+from:
+  - targetRef: # from allow-only-infra
+      kind: Mesh
+      name: mesh-1
+    conf:
+      action: DENY
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-monitoring
+    conf:
+      action: ALLOW
+  - targetRef: # from allow-only-infra
+      kind: MeshService
+      name: infra-logger
+    conf:
+      action: ALLOW
+  - targetRef: # from backend-permissions
+      kind: Mesh
+      name: mesh-1
+    conf:
+      action: ALLOW
+  - targetRef: # from backend-permissions
+      kind: MeshServiceSubset
+      name: web
+      tags:
+        version: v1
+    conf:
+      action: DENY
+```
+
+Rule-based view:
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshServiceSubset
+      name: web
+      tags:
+        version: v1
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshService
+      name: infra-monitoring
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshService
+      name: infra-logger
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: Mesh
+      name: mesh-1
+    conf:
+      action: ALLOW
+```
+
+#### ProxyTemplate
+
+ProxyTemplate is neither inbound nor outbound policy, 
+so it doesn't have "from" or "to" arrays. 
+There are only "imports" and "modifications" arrays
+which override arrays from other levels.
+That's why no merging is available for this policy
+(and this is a good thing taking into account complexity):
+
+```yaml
+type: ProxyTemplate
+mesh: default
+name: pt-1
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  conf:
+    imports:
+      - default-proxy
+    modifications:
+      - cluster:
+          operation: add
+          value: |
+            name: test-cluster
+            connectTimeout: 5s
+            type: STATIC
+```
+
+Now if you create a ProxyTemplate for DPP with an empty `modifications` array,
+`test-cluster` won't be added:
+
+```yaml
+type: ProxyTemplate
+mesh: default
+name: pt-2
+spec:
+  targetRef:
+    kind: MeshServiceSubset
+    name: backend
+    tags:
+      instance: my-special-backend-dpp
+  conf:
+    imports:
+      - default-proxy
+    modifications: [] # overrides "modifications" array from "pt-1"
+```
+
+#### TrafficLog
+
+Keep backends in the Mesh for now,
+move them into TrafficLogBackend policy in the future.
+
+Traffic log is a policy that contains both "to" and "from" arrays:
+```yaml
+type: TrafficLog
+mesh: mesh-1
+name: tl-1
+targetRef:
+  kind: Mesh
+  name: mesh-1
+to:
+  - targetRef:
+      kind: MeshService
+      name: web
+    conf:
+      backends:
+        - name: logstash
+from:
+  - targetRef:
+      kind: Mesh
+      name: mesh-1
+    conf:
+      backends:
+        - name: file
+```
+
+This policy specifies the following behaviour:
```

Comment:
so if I wanted to log traffic between service web-1 and web-2 on the web1 outbound I'd do:

```yaml
type: TrafficLog
mesh: mesh-1
name: tl-1
targetRef:
  kind: MeshService
  name: web-1
to:
  - targetRef:
      kind: MeshService
      name: web-2
    conf:
      backends:
        - name: file
```

right?

Reply:
Yes, this is correct